### PR TITLE
[plugin.video.roosterteeth] 1.4.1

### PR DESCRIPTION
**Pull request by skipmodea1**
### Description
1.4.1 (2019-06-09)
- added adapted inputstream switch. When this is on, the videoplayer inputstream addon 'Inputstream Adaptive' (kudo's to peak3d) is used. The addon 'Inputstream Adaptive' determines what resolution it will display based upon the available bandwidth and the settings of the addon 'Inputstream Adaptive'.
When the adapted inputstream switch is off, the roosterteeth addon will try and play the video based upon the maximum video resolution in the roosterteeth settings.
- for when the adapted inputstream switch is off: if a video of the desired video is not found, look for a video of lower quality
- fixed inside-gaming section
- added pages instead of only listing the first 30 items
- removed game attack
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        self.plugin_handle = int(sys.argv[1])
+
+        # log("ARGV", repr(sys.argv))
+        #
+        # Parse parameters...
+        self.url = urllib.parse.parse_qs(urllib.parse.urlparse(sys.argv[2]).query)['url'][0]
+        self.next_page_possible = urllib.parse.parse_qs(urllib.parse.urlparse(sys.argv[2]).query)['next_page_possible'][0]
+        self.show_serie_name = urllib.parse.parse_qs(urllib.parse.urlparse(sys.argv[2]).query)['show_serie_name'][0]
+
+        log("self.url", self.url)
```
Comment (Rechi) [inline plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py]: Kodi logs the url

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+            duration_in_seconds = length
+
+            studio = channel_slug
+            studio = convertToUnicodeString(studio)
+            studio = studio.replace("-", " ")
+            studio = studio.capitalize()
+
+            # Add to list...
+            list_item = xbmcgui.ListItem(label=title, thumbnailImage=thumbnail_url)
```
Comment (Rechi) [inline plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py]: thumbnailImage is deprecated use setArt

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            # Add our item to the listing as a 3-element tuple.
+            listing.append((plugin_url_with_parms, list_item, is_folder))
+
+        # Make a next page item, if a next page is possible
+        total_pages_str = json_data['total_pages']
+        total_pages = int(total_pages_str)
+        if self.page_number_next <= total_pages:
+            # Next page entry
+            if self.next_page_possible == 'True':
+                list_item = xbmcgui.ListItem(LANGUAGE(30200), thumbnailImage=os.path.join(IMAGES_PATH, 'next-page.png'))
```
Comment (Rechi) [inline plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py]: thumbnailImage is deprecated use setArt

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_serie_seasons.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            thumb = self.thumbnail_url
+
+            title = season_title
+
+            url = serie_url
+
+            thumbnail_url = thumb
+
+            # Add to list...
+            list_item = xbmcgui.ListItem(label=title, thumbnailImage=thumbnail_url)
```
Comment (Rechi) [inline plugin.video.roosterteeth/resources/lib/roosterteeth_list_serie_seasons.py]: thumbnailImage is deprecated use setArt

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_series.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            thumb = item['included']['images'][0]['attributes']['thumb']
+
+            title = serie_title
+
+            url = serie_url
+
+            thumbnail_url = thumb
+
+            # Add to list...
+            list_item = xbmcgui.ListItem(label=title, thumbnailImage=thumbnail_url)
```
Comment (Rechi) [inline plugin.video.roosterteeth/resources/lib/roosterteeth_list_series.py]: thumbnailImage is deprecated use setArt

File: plugin.video.roosterteeth/addon.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+from roosterteeth_const import ADDON, SETTINGS, LANGUAGE, IMAGES_PATH, DATE, VERSION
+
+# Parse parameters...
+if len(sys.argv[2]) == 0:
+    #
+    # Main menu
+    #
+    xbmc.log("[ADDON] %s, Python Version %s" % (ADDON, str(sys.version)), xbmc.LOGDEBUG)
+    xbmc.log("[ADDON] %s v%s (%s) is starting, ARGV = %s" % (ADDON, VERSION, DATE, repr(sys.argv)),
```
[the remark this answers is not in the document]
Comment (skipmodea1) [reply plugin.video.roosterteeth/addon.py]: Fixed

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py (hunk 1 above)
Comment (skipmodea1) [reply plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py]: Fixed

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py (hunk 2 above)
Comment (skipmodea1) [reply plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py]: Fixed

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py (hunk 3 above)
Comment (skipmodea1) [reply plugin.video.roosterteeth/resources/lib/roosterteeth_list_episodes.py]: Fixed

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_series.py (hunk 5 above)
Comment (skipmodea1) [reply plugin.video.roosterteeth/resources/lib/roosterteeth_list_series.py]: Fixed

File: plugin.video.roosterteeth/resources/lib/roosterteeth_list_serie_seasons.py (hunk 4 above)
Comment (skipmodea1) [reply plugin.video.roosterteeth/resources/lib/roosterteeth_list_serie_seasons.py]: Fixed

**Pull request merged**